### PR TITLE
Support windows emulation

### PR DESCRIPTION
**Pull request by sajal50**

### CODE REVIEW

File: generate.py [hunk 1]
```diff
@@ -29,53 +30,75 @@ def get_cmake_os(platform):
...
+            flags += configuration.get('cflags_c', [])
+        elif self._category == 'cc':
+            flags += configuration.get('cflags_cc', [])
+        else:
+            raise RuntimeError('Unknown category: ' + self._category)
+        flags += configuration.get('cflags', [])
+
+        return flags
+
+    @classmethod
```
Comment (frutiger) [inline generate.py]: Thanks for these, had forgotten to add them.

File: generate.py [hunk 2]
```diff
@@ -29,53 +30,75 @@ def get_cmake_os(platform):
...
+        elif self._category == 'cc':
+            flags += self._settings.GetCflagsCC(configuration_name)
+        else:
+            raise RuntimeError('Unknown category: ' + self._category)
+        flags += self._settings.GetCflags(configuration_name)
+
+        return flags
+
+    def defines(self, configuration_name, configuration):
+        defines = Properties.defines(configuration_name, configuration)
```
Comment (frutiger) [inline generate.py:81]: I believe `super` is preferable in case the hierarchy changes?

Comment (sajal50) [reply generate.py:81]: I actually not sure which is more Pythonic. But seems like we should be using `@staticmethod` instead?

Either way, does `super` make sense? The way I see this code is that it is using some utility which _happens_ to be in `super` class namespace.

Comment (frutiger) [reply generate.py:81]: I see. We can take that approach, but then I think we should remove the parent/child class relationship and forward all the methods explicitly.

**Review by frutiger — APPROVED** [review]

**Pull request merged**